# Pandas and plotting using matplotlib - 01

The purpose of this notebook is to practice simple data cleaning and manipulation using pandas; as well as basic plotting using matplotlib.

## Libraries

---

We start by importing the libraries to use

In [1]:
import pandas as pd
import numpy as np
import kagglehub
import re
import matplotlib.pyplot as plt
import matplotlib.dates as mdates
import matplotlib.ticker as mtick
import math

c:\Users\ossia\AppData\Local\Programs\Python\Python313\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


## The dataset to use

---

To practice we'll use the dataset [Melbourne Housing Snapshot](https://www.kaggle.com/datasets/dansbecker/melbourne-housing-snapshot), ranked number one for learning datasets in kaggle. To practice using the kagglehub API. We do this next:

In [ ]:
# kagglehub.dataset_download("dansbecker/melbourne-housing-snapshot", output_dir="../INPUTS/", force_download=True)

100%|██████████| 451k/451k [00:00<00:00, 2.88MB/s]

Extracting files...


'../INPUTS/'

> Running the previous command was a pain. Apparently, the .dataset_download function does not download to non-empty folders without the argument `force_dowload=True`.

### Notes on Specific Variables

- Rooms: Number of rooms
- Price: Price in dollars
- Method: S - property sold; SP - property sold prior; PI - property passed in; PN - sold prior not disclosed; SN - sold not disclosed; NB - no bid; VB - vendor bid; W - withdrawn prior to auction; SA - sold after auction; SS - sold after auction price not disclosed. N/A - price or highest bid not available.
- Type: br - bedroom(s); h - house,cottage,villa, semi,terrace; u - unit, duplex; t - townhouse; dev site - development site; o res - other residential.
- SellerG: Real Estate Agent
- Date: Date sold
- Distance: Distance from CBD
- Regionname: General Region (West, North West, North, North east …etc)
- Propertycount: Number of properties that exist in the suburb.
- Bedroom2 : Scraped # of Bedrooms (from different source)
- Bathroom: Number of Bathrooms
- Car: Number of carspots
- Landsize: Land Size
- BuildingArea: Building Size
- CouncilArea: Governing council for the area

## Reading the data

---

We now read the data we downloaded previously, and print its basic info

In [16]:
df_main = pd.read_csv("../INPUTS/melb_data.csv")
display(df_main.head(5))

,Suburb,Address,Rooms,Type,Price,Method,SellerG,Date,Distance,Postcode,...,Bathroom,Car,Landsize,BuildingArea,YearBuilt,CouncilArea,Lattitude,Longtitude,Regionname,Propertycount
0,Abbotsford,85 Turner St,2,h,1480000.0,S,Biggin,3/12/2016,2.5,3067.0,...,1.0,1.0,202.0,NaN,NaN,Yarra,-37.7996,144.9984,Northern Metropolitan,4019.0
1,Abbotsford,25 Bloomburg St,2,h,1035000.0,S,Biggin,4/02/2016,2.5,3067.0,...,1.0,0.0,156.0,79.0,1900.0,Yarra,-37.8079,144.9934,Northern Metropolitan,4019.0
2,Abbotsford,5 Charles St,3,h,1465000.0,SP,Biggin,4/03/2017,2.5,3067.0,...,2.0,0.0,134.0,150.0,1900.0,Yarra,-37.8093,144.9944,Northern Metropolitan,4019.0
3,Abbotsford,40 Federation La,3,h,850000.0,PI,Biggin,4/03/2017,2.5,3067.0,...,2.0,1.0,94.0,NaN,NaN,Yarra,-37.7969,144.9969,Northern Metropolitan,4019.0
4,Abbotsford,55a Park St,4,h,1600000.0,VB,Nelson,4/06/2016,2.5,3067.0,...,1.0,2.0,120.0,142.0,2014.0,Yarra,-37.8072,144.9941,Northern Metropolitan,4019.0


## EDA

---

Ahora hacemos un analisis exploratorio *muy* superficial del dataframe; empezando con el uso del metodo `.info()`.

In [17]:
display(df_main.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  float64
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  float64
 10  Bedroom2       13580 non-null  float64
 11  Bathroom       13580 non-null  float64
 12  Car            13518 non-null  float64
 13  Landsize       13580 non-null  float64
 14  BuildingArea   7130 non-null   float64
 15  YearBuilt      8205 non-null   float64
 16  CouncilArea    12211 non-null  object 
 17  Lattitude      13580 non-null  float64
 18  Longti

None

We can see from the results that the dataframe has 13,580 entries, 21 columns, and only the following columns have null values:

- Car
- BuildingArea
- YearBuilt
- CouncilArea

## Questions to answer.

1. What percentage of the market 

## **PENDIENTE**

## Initial Assesment - High level questions.

To avoid just doing braindead data manipulation and inconsecuantial graphs, we first need make an initial assesment of the dataset. We need to pose high level questions and define a line of inquiry.

### What is the data about?

The data is about the housing market in Melbourne, Australiia. It contains mostly information about the attributes/characteristic of properties that have circulated the market.

### Lines of inquiry

For now, i do not plan on making a predictive analysis. We are stuck then with making merely descriptive assesments.